exchanges/okx/okx_wrapper.go

```diff
@@ -0,0 +1,1329 @@
+	if ok.Verbose {
+		log.Debugf(log.ExchangeSys,
+			"%s Websocket: %s.",
+			ok.Name,
+			common.IsEnabled(ok.Websocket.IsEnabled()))
+		ok.PrintEnabledPairs()
+	}
+
+	if !ok.GetEnabledFeatures().AutoPairUpdates {
+		return
+	}
+	err := ok.UpdateTradablePairs(context.TODO(), false)
+	if err != nil {
+		log.Errorf(log.ExchangeSys,
+			"%s failed to update tradable pairs. Err: %s",
+			ok.Name,
+			err)
+	}
+}
+
+// FetchTradablePairs returns a list of the exchanges tradable pairs
+func (ok *Okx) FetchTradablePairs(ctx context.Context, a asset.Item) ([]string, error) {
+	if !ok.SupportsAsset(a) {
+		return nil, fmt.Errorf("asset type of %s is not supported by %s", a, ok.Name)
+	}
+	format, err := ok.GetPairFormat(a, false)
+	if err != nil {
+		return nil, err
+	}
+	pairs := []string{}
+	insts := []Instrument{}
+	switch a {
+	case asset.Spot:
+		insts, err = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeSpot,
+		})
+	case asset.Futures:
+		insts, err = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeFutures,
+		})
+	case asset.PerpetualSwap:
+		insts, err = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeSwap,
+		})
+	case asset.Option:
+		var instsb []Instrument
+		var instsc []Instrument
+		insts, err = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeOption,
+			Underlying:     "BTC-USD",
+		})
+		if err != nil {
+			return nil, err
+		}
+		instsb, err = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeOption,
+			Underlying:     "ETH-USD",
+		})
+		if err != nil {
+			return nil, err
+		}
+		instsc, err = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeOption,
+			Underlying:     "SOL-USD",
+		})
+		insts = append(insts, instsb...)
+		insts = append(insts, instsc...)
+	case asset.Margin:
+		insts, err = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeMargin,
+		})
+	}
+	if err != nil {
+		return nil, err
+	}
+	if len(insts) == 0 {
+		return nil, errNoInstrumentFound
+	}
+	for x := range insts {
+		var pair string
+		switch insts[x].InstrumentType {
+		case asset.Spot, asset.Margin:
+			pair = insts[x].BaseCurrency + format.Delimiter + insts[x].QuoteCurrency
+			if pair == "-" {
+				return nil, fmt.Errorf("%v, invalid currency pair data", errMalformedData)
+			}
+		case asset.Futures, asset.PerpetualSwap, asset.Option:
+			c, err := currency.NewPairFromString(insts[x].InstrumentID)
+			if err != nil {
+				return nil, err
+			}
+			pair = c.Base.String() + format.Delimiter + c.Quote.String()
+		}
+		pairs = append(pairs, pair)
+	}
+	selectedPairs := []string{}
+	pairsMap := map[string]int{}
+	for i := range pairs {
+		if pairs[i] == "" {
+			return nil, errInvalidCurrencyPair
+		}
+		count, ok := pairsMap[pairs[i]]
+		if !ok || count == 0 {
+			pairsMap[pairs[i]] = 1
+			selectedPairs = append(selectedPairs, pairs[i])
+		}
+	}
```

Comment:
Can be done in the range above, can also use map[string]bool for more concise code.